pkg/inject/inject.go

```diff
@@ -384,15 +384,65 @@ func (conf *ResourceConfig) CreateDefaultOpaquePortsPatch() ([]byte, error) {
 			return nil, err
 		}
 	} else if !conf.HasWorkloadAnnotation(k8s.ProxyOpaquePortsAnnotation) {
-		opaquePorts := conf.GetValues().Proxy.OpaquePorts
-		patch, err = conf.CreateAnnotationPatch(opaquePorts)
+		defaultPorts := strings.Split(conf.GetValues().Proxy.OpaquePorts, ",")
+		var filteredPorts []string
```

Comment:
Definitely _very_ nitpicky, but perhaps this function can be simplified a bit? It's not necessarily hard to follow, but the nesting level does make it a bit more difficult to understand.

wdyt about inverting some branches and doing away with some redundant error checking? For example on L382 -- we error check but then return the error to the caller if it is not nil anyway (presumably the caller will check the error again), we could instead just return straight away there.
```go
func (conf *ResourceConfig) CreateDefaultOpaquePortsPatch() ([]byte, error) {
  if opaquePorts, ok := conf.GetConfigAnnotation(k8s.ProxyOpaquePortsAnnotation); ok {
       // Annotation is present on namespace
      // let caller deal with the error
     return conf.CreateAnnotationPatch(opaquePorts)
  }

  var patch []byte
  var err error
  if conf.HasWorkloadAnnotation(...) {
       // annotation is already present, return (nil, nil)
      // helps us reduce the nesting
     return patch, err
  }

  // past this point everything would stay the same
```

This might make it easier to follow? Although not a blocker from my side; also, explicit error checking has its uses and having just one return at the end (where we return the patch) is definitely a good way to reason with what's happening. Also, this might get changed again soon so not sure if it's worth really spending a lot of time on it 🤷🏻

Reply:
I reduced some of the nesting and changed the name to `CreateOpaquePortsPatch` since it does not necessarily create a default list; the workload may already the annotation or it may inherit one from its namespace.